handlers/authentication.go

```diff
@@ -285,9 +279,37 @@ func performOpenIdAuthentication(w http.ResponseWriter, r *http.Request) bool {
 		return false
 	}
 
-	// Create business layer using the received id_token
-	expiresOn := time.Now().Add(time.Second * time.Duration(expiresInNumber))
+	// Set a default value for expiration date
+	expiresOn := time.Now().Add(time.Second * time.Duration(defaultSessionDuration))
+
+	// We check for the expiration date from the claim
+	if expClaim := idTokenClaims["exp"].(string); expClaim != "" {
+		expiresInNumber, err := strconv.Atoi(expClaim)
+
+		if err != nil {
+			RespondWithDetailedError(w, http.StatusBadRequest, "Token exp claim is present, but invalid.", err.Error())
+			return false
+		}
+
+		expiresOn = time.Now().Add(time.Second * time.Duration(expiresInNumber))
+	}
 
+	// Then we check the expiration date for the access on the access token
+	// We give priority to it because it is what we use to access the cluster,
+	// so if it is valid we are ok.
```

Comment:
@mikolajwrzesinski has already noted that the OpenId spec tells that the `expires_in` parameter is tied to the `access_token` (see https://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthResponse).

For OpenID we are *not* using the `access_token`, but the `id_token`. We can safely ignore `expires_in` parameter. So, this block should be removed.

Reply:
Since we do use the `access_token` to to the requests to kubernetes, wouldn't it be the case where the expiration provided on `expires_in` be more reliable than the exp on the id_token?

Reply:
The `id_token` is the one used for OpenID (see the checks at https://github.com/kiali/kiali/pull/2981/files#diff-7b14eebb8414820b40c718efcf921867R244-R249). The `access_token` is ignored in this case. So, `expires_in` is not that reliable.

For OpenShift it is different. In OpenShift, we DO use the `access_token` and it's better to use `expires_in`. But for this PR, we should move to using the `exp` claim.

Reply:
See also an already fixed issue: https://github.com/kiali/kiali/issues/2925. It's mentioned that some IdPs won't send an `access_token`.